src/Calculator.elm

```diff
@@ -0,0 +1,150 @@
+module Counter exposing (Model, Msg, init, update, view)
+
+import Browser
+import Browser.Events
+import Html exposing (Attribute, Html, div, input, text)
+import Html.Events exposing (onClick, onInput)
+import Json.Decode
+import String exposing (dropRight, length, toFloat, toInt)
+
+
+main =
+    Browser.element { init = init, update = update, view = view, subscriptions = subscriptions }
+
+
+type alias Model =
+    { value : String, secondNumber : Float }
+
+
+type alias Flags =
+    ()
+
+
+type Msg
+    = ChangedAddText String
+    | KeyDown String
+    | Add
+    | Subtract
+    | Multiply
+    | Divide
+    | Power
+    | Sqrt
+
+
+parseNumber : String -> Float
+parseNumber text =
+    let
+        theMaybe =
+            toFloat text
+    in
+    case theMaybe of
+        Just val ->
+            val
+
+        Nothing ->
+            0
+
+
+typeNumber : String -> String -> String
+typeNumber value key =
+    let
+        numberToAdd =
+            toInt key
+    in
+    if key == "Backspace" then
+        if length value == 1 then
+            "0"
+
+        else
+            dropRight 1 value
+
+    else if key == "Enter" then
+        "0"
+
+    else if key == "." && Maybe.withDefault 0.5 (toFloat value) / 1 == 0 then
+        value ++ "."
+
+    else
+        case numberToAdd of
+            Nothing ->
+                value
+
+            Just number ->
+                if not (value == "0") then
+                    value ++ String.fromInt number
+
+                else
+                    String.fromInt number
+
+
+init : Flags -> ( Model, Cmd Msg )
+init _ =
+    ( { value = "0", secondNumber = 0 }, Cmd.none )
```

Comment:
If you pass around "empty" data (like ignoring your argument or returning a tuple with a default value like `Cmd.none` or `0`), it's a sign your function should have a different signature.
In this case:
```elm
init : Model
init =
    { value = "0"
    , secondNumber = 0
    }
```
and then in `main`:
```elm
    Browser.element
        { init = always (init, Cmd.none)
        …
```
Notice how we put all uninteresting "glue" just where it's needed?
This way, you can much more easily reuse `init`, because you don't need to strip the interesting part away.